### PR TITLE
Add random_seed parameter

### PR DESCRIPTION
**Pull request by ArnoldYSYeung**
Allows user to input a random_seed for repeating runs. If None, does not use random_seed. 
Code has been fixed to apply to every instance of np.random and rd.random. (Seed must be defined prior to every use of np.random and rd.random.)

Code has been tested on `smogn/examples/smogn_example_1_beg.ipynb` . Using the same `random_seed` value always output the same stats.

```
housing_smogn = smogn.smoter(
    
    data = housing,  ## pandas dataframe
    y = 'SalePrice',  ## string ('header name')
    random_seed = 1
)
smogn.box_plot_stats(housing_smogn['SalePrice'])['stats']
```

### CODE REVIEW

**Comment by ArnoldYSYeung** [comment]
@nickkunz The modifications here have been tested with the `smogn/examples/smogn_example_1_beg.ipynb` notebook. Please test with more complex cases as well. Thanks.

**Comment by Muti23** [comment]
@ArnoldYSYeung Thanks for your sharing. In fact I also used the same strategy to deal with my problem. I try to implant random seed for several rows in smoter/over_sampling files. Unfortunately it doesn't  solve the problem completely, and only increase the reproducibility of test results. I don't know if you have the same situation? And I will use your modifications to do the same test.

**Comment by polinamamo** [comment]
@Muti23 my understanding after reading the code, that you need to fix seed for both `np.random.seed(seed)`
    `rd.seed(seed)` at the same time, as both methods are used for random sampling.

**Review by nickkunz — CHANGES_REQUESTED** [review]
@ArnoldYSYeung can you remove the locally generated pycache files?

File: smogn/over_sampling.py [hunk 1]
```diff
@@ -12,12 +12,12 @@
...
-    perc,       ## over / under sampling
-    pert,       ## perturbation / noise percentage
-    k           ## num of neighs for over-sampling
-    
+    data,               ## training set
+    index,              ## index of input data
+    perc,               ## over / under sampling
+    pert,               ## perturbation / noise percentage
+    k,                  ## num of neighs for over-sampling
+    random_seed = None  ## random seed for sampling (pos int or None)
```
Comment (nickkunz) [inline smogn/over_sampling.py:20]: @ArnoldYSYeung are you open to changing the `random_seed` argument to simply `seed` for brevity?

**Comment by nickkunz** [comment]
@ArnoldYSYeung @Muti23 @polinamamo thank you all for your input. This is very helpful for everyone using the package.

**Comment by ArnoldYSYeung** [comment]
@nickkunz Renamed `random_seed` to `seed` and removed pycache files. Please take a look and let me know of any other changes 🙏

**Comment by nickkunz** [comment]
@ArnoldYSYeung Can we merge this to the `dev` branch before we merging to `master`?

**Review by nickkunz — APPROVED** [review]

**Review by nickkunz — CHANGES_REQUESTED** [review]
@ArnoldYSYeung Can we merge this to the dev branch before we merging to master?

**Comment by ArnoldYSYeung** [comment]
@nickkunz Changed base branch from `master` to `dev`

**Review by nickkunz — APPROVED** [review]

**Comment by nickkunz** [comment]
@ArnoldYSYeung Thank you. I really appreciate your contribution. This needed to be done.

**Pull request merged**